src/ConnectedMode.UnitTests/BranchMatcherTests.cs

```diff
@@ -236,13 +236,13 @@ public async Task GetMatchedBranch_NoMatchingBranch_ChooseMain()
         }
 
         [TestMethod]
-        public async Task GetMatchedBranch_RepoHasNoHead_ReturnsMainBranch()
+        public async Task GetMatchedBranch_RepoHasNoHead_ReturnsNull()
         {
             var testSubject = CreateTestSubject(CreateSonarQubeService("any").Object);
 
             var result = await testSubject.GetMatchingBranch("projectKey", Mock.Of<IRepository>(), CancellationToken.None);
 
-            result.Should().Be("any");
+            result.Should().BeNull();
```

Comment:
The name and behaviour of this test doesn't match the description or the description in the bug, both of which seem to say we should return [main] for headless repos. Which is correct?